include/readdy/model/observables/ObservableFactory.h

```diff
@@ -59,51 +59,145 @@ template<typename T>
 using is_observable_type = std::enable_if_t<std::is_base_of<model::observables::ObservableBase, T>::value>;
 }
 
+// fixme: adress the following points:
+// 1. Not satisfied with the overloading combinatorial explosion for default arguments, go fully explicit?
+// 2. should callback functions always take a cref as argument.
```

Comment:
minimal overhead as it's all function pointers

Reply:
I meant the argument of the actual callback function, i.e. either `[](const auto& result){}` or `[](auto result){}`

Reply:
so if i recall recorrectly pybind supports cref so it should be a cref